connection.go

```diff
@@ -648,3 +648,10 @@ func (mc *mysqlConn) ResetSession(ctx context.Context) error {
 func (mc *mysqlConn) IsValid() bool {
 	return !mc.closed.Load()
 }
+
+func (mc *mysqlConn) errLog() Logger {
+	if mc.cfg.Logger != nil {
+		return mc.cfg.Logger
```

Comment:
It seems ugly. Can you make `mc.cfg.Logger` always alive?

Reply:
I agree.
That is possible by set the default logger to `Config.Logger` on `NewConfig()` and `Config.normalize()`, but this way has a small impact on `SetLogger()` behavior.
In this case, existing connections will continue to use the default logger even after calling `SetLogger()`.
However, it seems generally not good to call `SetLogger()` after creating connections, so maybe I shouldn't worry too much about this change.

What do you think about this?

Reply:
+1

Reply:
I'll fix as above.